Apps/WebClient/src/ClientApp/src/stores/delegate.ts

```diff
@@ -0,0 +1,125 @@
+import { ResultError } from "@/models/errors";
+import { CreateDelegateInvitationRequest } from "@/models/sharing/createDelegateInvitationRequest";
+import { DelegateInvitation } from "@/models/sharing/delegateInvitation";
+import { IDelegateService, ILogger } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+
+export enum DelegateInvitationDialogStep {
+    contact = 0,
+    nickname = 1,
+    dataSources = 2,
+    expiryDate = 3,
+    review = 4,
+    sharingCode = 5,
+}
+
+export interface DelegateInvitationDialogState {
+    mode: "Create" | "Modify";
+    step: DelegateInvitationDialogStep;
+    nickname?: string;
+    email?: string;
+    expiryDate?: string;
+    dataSources?: DataSource[];
+}
+
+const defaultInvitationDialogState: DelegateInvitationDialogState = {
+    mode: "Create",
+    step: DelegateInvitationDialogStep.contact,
+};
+
+export const useDelegateStore = defineStore("delegate", () => {
+    const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    const delegateService = container.get<IDelegateService>(
+        SERVICE_IDENTIFIER.DelegateService
+    );
+
+    const errorStore = useErrorStore();
+
+    const invitations = ref<DelegateInvitation[]>([]);
+    const invitationDialogState = ref<DelegateInvitationDialogState>({
+        ...defaultInvitationDialogState,
+    });
+    const invitationsAreLoading = ref(false);
+    const error = ref<ResultError>();
+    const statusMessage = ref("");
+
+    function setInvitationsError(incomingError: ResultError) {
+        error.value = incomingError;
+        statusMessage.value = incomingError.resultMessage;
+        invitationsAreLoading.value = false;
+    }
+
+    function submitInvitationDialog(): Promise<string | void> {
+        if (invitationDialogState.value.mode === "Create") {
+            return createInvitation({
+                status: DelegateInvitationStatus.Pending,
+                nickname: invitationDialogState.value.nickname,
+                email: invitationDialogState.value.email,
+                expiryDate: invitationDialogState.value.expiryDate,
+                dataSources: invitationDialogState.value.dataSources,
+            });
+        }
+        // TODO: Remove this once all paths are covered
+        return Promise.resolve();
+    }
+
+    function createInvitation(
+        invitation: CreateDelegateInvitationRequest
+    ): Promise<string | void> {
+        logger.debug("Create invitation");
+        invitationsAreLoading.value = true;
+        return delegateService
+            .createInvitation(invitation)
+            .catch((error: ResultError) => {
+                handleError(error, ErrorType.Create, "sharing-dialog");
+            })
+            .then((result) => {
+                // TODO: retrieve all invitations, when invitation is successfully created
+                invitationsAreLoading.value = false;
+                resetInvitationDialogState();
+                return result;
```

Comment:
Technically, the dialog is not done after an invitation is created, because there is an additional page where the sharing code is displayed.  Instead of resetting the state, I think we should move the sharingCode into DelegateInvitationDialogState, then the state can be updated so as to display the new page and the code won't need to be returned.

Reply:
I've made some changes but any further changes at this point will be too much optimization before the implementation, I'll keep this in mind, the recent change already aligns to this so idea so it should follow through into the implementation.